app/lib/fake/server/fake_server_entrypoint.dart

```diff
@@ -66,14 +66,7 @@ class FakeServerCommand extends Command {
     final dataFile = argResults!['data-file'] as String?;
     final watch = argResults!['watch'] == true;
 
-    Logger.root.onRecord.listen((r) {
-      print([
-        r.time.toIso8601String(),
-        r.toString(),
-        r.error,
-        r.stackTrace?.toString(),
-      ].where((e) => e != null).join(' '));
-    });
+    setupDebugEnvBasedLogging(defaultDebugEnvValue: 'fake_server pub.email');
```

Comment:
This is because integration tests wouldn't work without this hack?

In that case, we could:
 * add a comment saying that we need to always have this by default for integration tests to pass.
 * We could specify `DEBUG=...` when running integration tests, since the integration tests actually launch the fake_server_entrypoint, right?

Specifically, why don't we just in:
https://github.com/dart-lang/pub-dev/blob/fee7f33c94bce8607eba24a6a80ba289d76309d5/pkg/pub_integration/lib/src/fake_pub_server_process.dart#L49-L67

specify the env var `DEBUG`, indeed we can specify it as:
```
  'DEBUG': 'fake_server pub.email ' + Platform.environment['DEBUG'] ?? ''
```

since it'll split with whitespace.

Reply:
Updated with `DEBUG` defaulting to `*` for locally running it.